kernel/kernbench.py

```diff
@@ -84,21 +84,35 @@ def setUp(self):
         Setting up the env for the kernel building
         """
         smg = SoftwareManager()
-        detected_distro = distro.detect()
+        self.detected_distro = distro.detect()
         deps = ['gcc', 'make', 'automake', 'autoconf', 'time', 'bison', 'flex']
-        if 'Ubuntu' in detected_distro.name:
+        if 'Ubuntu' in self.detected_distro.name:
             deps.extend(['libpopt0', 'libc6', 'libc6-dev', 'libpopt-dev',
                          'libcap-ng0', 'libcap-ng-dev', 'elfutils', 'libelf1',
                          'libnuma-dev', 'libfuse-dev', 'libssl-dev'])
-        elif 'SuSE' in detected_distro.name:
+        elif 'SuSE' in self.detected_distro.name:
             deps.extend(['libpopt0', 'glibc', 'glibc-devel',
                          'popt-devel', 'libcap2', 'libcap-devel',
                          'libcap-ng-devel', 'openssl-devel'])
-        elif detected_distro.name in ['centos', 'fedora', 'rhel']:
+        elif self.detected_distro.name in ['centos', 'fedora', 'rhel']:
             deps.extend(['popt', 'glibc', 'glibc-devel', 'libcap-ng',
                          'libcap', 'libcap-devel', 'elfutils-libelf',
                          'elfutils-libelf-devel', 'openssl-devel'])
 
+        if 'rhel' in self.detected_distro.name:
+            '''
+            In Rhel Based Distro kernel build is failing as system config file
+            has crypto module . This work around disabled those so it allow to
+            build kernel
+            '''
+            process.system("cp /boot/config* /tmp", shell=True, sudo=True)
```

Comment:
Just one suggestion, better to copy file with unique filenames and remove the same so that we don't touch other files with `config*`. Otherwise looks good.